test/python/compiler/test_assembler.py

```diff
@@ -1224,6 +1224,108 @@ def test_assemble_single_instruction(self):
         qobj = assemble(inst, self.backend)
         validate_qobj_against_schema(qobj)
 
+    def test_assemble_qubits_not_in_meas_map(self):
+        """Testing that the assembly errors when measured qubits not in meas_map."""
+        schedule = Schedule()
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(0), MemorySlot(0)),
+        )
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(1), MemorySlot(1)) << 1,
+        )
+        schedule = schedule.append(
+            Acquire(10, AcquireChannel(3), MemorySlot(3)) << 1,
+        )
+        with self.assertRaises(QiskitError):
+            assemble(schedule,
+                     qubit_lo_freq=self.default_qubit_lo_freq,
+                     meas_lo_freq=self.default_meas_lo_freq,
+                     meas_map=[[0, 1]])
+
+    def test_assemble_overlapping_time(self):
+        """Test that assembly errors when qubits are measured in overlapping time."""
+        schedule = Schedule()
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(0), MemorySlot(0)),
+        )
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(1), MemorySlot(1)) << 1,
+        )
+        with self.assertRaises(QiskitError):
+            assemble(schedule,
+                     qubit_lo_freq=self.default_qubit_lo_freq,
+                     meas_lo_freq=self.default_meas_lo_freq,
+                     meas_map=[[0, 1]])
+
+    def test_assemble_non_overlapping_time_single_meas_map(self):
+        """Test that assembly works when qubits are measured in non-overlapping
+        time within the same measurement map list."""
+        schedule = Schedule()
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(0), MemorySlot(0)),
+        )
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(1), MemorySlot(1)) << 5,
+        )
+        qobj = assemble(schedule,
+                        qubit_lo_freq=self.default_qubit_lo_freq,
+                        meas_lo_freq=self.default_meas_lo_freq,
+                        meas_map=[[0, 1]])
+        validate_qobj_against_schema(qobj)
+
+    def test_assemble_disjoint_time(self):
+        """Test that assembly works when qubits are measured in disjoint time."""
+        schedule = Schedule()
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(0), MemorySlot(0)),
+        )
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(1), MemorySlot(1)) << 1,
+        )
+        qobj = assemble(schedule,
+                        qubit_lo_freq=self.default_qubit_lo_freq,
+                        meas_lo_freq=self.default_meas_lo_freq,
+                        meas_map=[[0, 2], [1, 3]])
+        validate_qobj_against_schema(qobj)
+
+    def test_assemble_invalid_qubits(self):
```

Comment:
duplicate of `test_assemble_qubits_not_in_meas_map`

Reply:
fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/ac87c70236ab323f937d1fa4e4ce78a67ec39e51